### PR TITLE
fix issue of user defined kernel

### PR DESCRIPTION
**Pull request by anaruse**
Fixed the issue of user defined kernel with fully manual indexing. #1802

### CODE REVIEW

File: cupy/core/elementwise.pxi [hunk 1]
```diff
@@ -365,12 +365,13 @@ cdef list _get_out_args(list out_args, tuple out_types, tuple out_shape,
...
 
 cdef list _get_out_args_with_params(
-        list out_args, tuple out_types, tuple out_shape, tuple out_params):
+        list out_args, tuple out_types, tuple out_shape, tuple out_params, size):
     cdef ParameterInfo p
     if not out_args:
         for p in out_params:
             if p.raw:
-                raise ValueError('Output array size is Undecided')
+                if size is not None:
```
Comment (okuta) [inline cupy/core/elementwise.pxi]: How about `if p.raw and size is not None`?

File: cupy/core/elementwise.pxi [hunk 2]
```diff
@@ -365,12 +365,13 @@ cdef list _get_out_args(list out_args, tuple out_types, tuple out_shape,
 
 
 cdef list _get_out_args_with_params(
-        list out_args, tuple out_types, tuple out_shape, tuple out_params):
+        list out_args, tuple out_types, tuple out_shape, tuple out_params, size):
```
Comment (okuta) [inline cupy/core/elementwise.pxi]: How about `tuple size=None`?

**Comment by okuta** [comment]
LGTM except comments.

**Comment by anaruse** [comment]
Thank you for your comments and sorry for my slow response (I had limited access since last Tue).  I found  stupid bug in my patch. I'll fix it and the code parts you mentioned and add testing script later.

File: cupy/core/elementwise.pxi (hunk 2 above)
Comment (anaruse) [inline cupy/core/elementwise.pxi]: When "tuple size=None" is used, I get the following error: 

    Traceback (most recent call last):
      File "test_user-kernel.py", line 27, in <module>
        out2 = uesr_kernel_2( in1.data, in2.data, size=n )
      File "cupy/core/elementwise.pxi", line 526, in cupy.core.core.ElementwiseKernel.__call__ (cupy/core/core.cpp:35984)
    TypeError: Expected tuple, got int

File: cupy/core/elementwise.pxi (hunk 1 above)
Comment (anaruse) [inline cupy/core/elementwise.pxi]: Sorry, this conditional expression I added was wrong (and opposite). Correct one is "if size is None:". will update including your comment.

File: tests/cupy_tests/core_tests/test_userkernel.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        uesr_kernel_2 = cupy.ElementwiseKernel(
+            'raw T x, raw T y',
+            'raw T z',
+            '''
+                z[i] = x[i] + y[i];
+            ''',
+            'uesr_kernel_2')
+        out2 = uesr_kernel_2(in1, in2, size=n)
+
+        for i in range(n):
```
Comment (okuta) [inline tests/cupy_tests/core_tests/test_userkernel.py]: Please use `cupy.testing.assert_array_equal`.

**Comment by anaruse** [comment]
Thank you for your comment.  Fix test code to use cupy.testing APIs.

**Pull request merged**